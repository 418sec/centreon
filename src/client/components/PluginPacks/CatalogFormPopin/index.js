export { default as CatalogFormPopin } from './CatalogFormPopinContainer';