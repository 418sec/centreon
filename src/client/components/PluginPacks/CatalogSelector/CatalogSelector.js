import React, { Component } from 'react'
import { withStyles } from 'material-ui/styles'
import Avatar from 'material-ui/Avatar';
import deepPurple from 'material-ui/colors/deepPurple'
import Error from 'material-ui-icons/Error'

const styles = {
  root: {
    display: "inline-block"
  },
  avatar: {
    width: 22,
    height: 22,
    fontSize: 14,
    cursor: "pointer"
  },
  purpleAvatar: {
    width: 22,
    height: 22,
    fontSize: 14,
    cursor: "pointer",
    backgroundColor: deepPurple[400],
  },
  errorIcon: {
    fill: '#d1d2d4',
    cursor: "pointer"
  },
}

class CatalogSelector extends Component {

  render = () => {
    const { classes, handleClick, level } = this.props

    if (level !== null) {
      return (
        <Avatar
          onClick={handleClick}
          className={classes.purpleAvatar}
        >
          {level}
        </Avatar>
      )
    } else {
      return (
        <div className={classes.root}>
          <Error
            onClick={handleClick}
            className={classes.errorIcon}
          />
        </div>
      )
    }
  }
}

export default withStyles(styles)(CatalogSelector)