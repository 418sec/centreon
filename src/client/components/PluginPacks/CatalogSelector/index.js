export { default as CatalogSelector } from './CatalogSelectorContainer';