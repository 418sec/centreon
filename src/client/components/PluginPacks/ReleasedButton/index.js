export { default as ReleasedButton } from './ReleasedButtonContainer';