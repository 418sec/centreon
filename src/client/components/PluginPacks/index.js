export { default as PluginPacksList } from './PluginPacksListContainer';