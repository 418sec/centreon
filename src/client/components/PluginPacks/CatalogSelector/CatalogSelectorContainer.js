import React, { Component } from 'react'
import { connect } from 'react-redux'
import CatalogSelector from './CatalogSelector'
import { openCatalogFormPopin } from 'Redux/pluginPacks/pluginPacksActions'

class CatalogSelectorContainer extends Component {

  handleClick = () => {
    const { id, level } = this.props
    this.props.openCatalogFormPopin(id, level)
  }

  render = () => {
    const { level } = this.props

    return (
      <CatalogSelector
        handleClick={this.handleClick}
        level={level}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {}
}


const mapDispatchToProps = (dispatch) => {
  return {
    openCatalogFormPopin: (pluginPackId, level) => {
      return dispatch(openCatalogFormPopin(pluginPackId, level))
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(CatalogSelectorContainer)