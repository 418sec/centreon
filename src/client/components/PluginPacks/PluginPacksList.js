import React, { Component } from 'react'
import {
  PagingState,
  FilteringState,
  CustomPaging
} from '@devexpress/dx-react-grid'
import {
  Grid,
  Table,
  TableHeaderRow,
  TableFilterRow,
  PagingPanel
} from '@devexpress/dx-react-grid-material-ui'
import PropTypes from 'prop-types'
import { withStyles } from 'material-ui/styles'
import { TableRow, TableCell } from 'material-ui'
import { HeaderBar } from 'Components/HeaderBar'
import { CatalogFormPopin } from 'Components/PluginPacks/CatalogFormPopin'
import { CatalogSelector } from 'Components/PluginPacks/CatalogSelector'
import { ReleasedButton } from 'Components/PluginPacks/ReleasedButton'
import { FilterCell } from 'Components/PluginPacks/FilterCell'

const styles = theme => ({
  root: {
    padding: '0 15px',
    overflow: 'auto'
  },
  TableCell: {
    wordWrap: 'break-word',
    padding: '4px 14px',
  },
  check: {
    fill: '#88B917'
  },
  checkInactive: {
    fill: '#d1d2d4'
  },
  textField: {
    '&:hover': {
      background:'transparent'
    },
  }
})

class PluginPacksList extends Component {

  tableRowTemplate = ({children, row}) => {
    return (
      <TableRow id={row.slug}>
        {children}
      </TableRow>
    )
  }

  render = () => {
    const { columns, rows, itemsPerPage, totalItems, currentPage, classes } = this.props

    return (
      <section>
        <div id="pluginPackList" className={classes.root}>
          <Grid
            columns={columns}
            rows={rows}
          >
            <FilteringState
              onFiltersChange={this.props.changeFilters}
            />
            <PagingState
              currentPage={currentPage}
              onCurrentPageChange={this.props.changeCurrentPage}
              pageSize={itemsPerPage}
            />
            <CustomPaging
              totalCount={totalItems}
            />
            <Table
              rowComponent={this.tableRowTemplate}
              cellComponent={({ column, row, value }) => {
                if (column.name === 'released') {
                  return (
                    <TableCell>
                      <ReleasedButton
                        pluginPackId={row.id}
                      />
                    </TableCell>
                  )
                } else if (column.name === 'catalog_level') {
                  return (
                    <TableCell>
                      <CatalogSelector
                        id={row.id}
                        level={value}
                      />
                    </TableCell>
                  )
                }

                return (
                  <TableCell className={classes.TableCell}>{value}</TableCell>
                )
              }}
            />
            <TableHeaderRow />
            <PagingPanel id={"pagingPanel"}/>
            <TableFilterRow
              messages={{ filterPlaceholder: 'Filter' }}
              cellComponent={FilterCell}
            />
          </Grid>
        </div>
        <CatalogFormPopin />
      </section>
    )
  }
}

/**
 * Instance list Prototypes
 */
PluginPacksList.propTypes = {
  /**
   * Instance list css classes
   * @param {Object} classes
   */
  classes: PropTypes.object.isRequired,
}

PluginPacksList.defaultProps = {
  currentPage: 1
}

export default withStyles(styles)(PluginPacksList)
