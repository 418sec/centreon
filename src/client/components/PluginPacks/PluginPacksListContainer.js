import React, { Component } from 'react'
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux'
import PluginPacksList from './PluginPacksList'
import { getPluginPacks } from 'Services/hubApi'
import { cleanHeader, replaceTitle} from 'Redux/header/headerActions'

class PluginPacksListContainer extends Component {

  constructor (props) {
    super(props)

    this.state = {
      columns: [
        {name: 'slug', title: 'Slug'},
        {name: 'name', title: 'Name'},
        {name: 'version', title: 'Version'},
        {name: 'catalog_level', title: 'Catalog'},
        {name: 'released', title: 'Released'},
      ],
      currentPage: 0,
      itemsPerPage: 10,
      filters: []
    }
  }

  componentDidMount = () => {
    const { cleanHeader, replaceHeaderTitle } = this.props
    cleanHeader()
    replaceHeaderTitle('Plugin Packs')

    this.getData()
  }

  changeFilters = (filters) => {
    this.setState(
      {
        filters: filters,
        currentPage: 0
      },
      this.getData
    )
  }

  changeCurrentPage = (currentPage) => {
    this.setState(
      {currentPage: currentPage},
      this.getData
    )
  }

  /**
   * example : page[number]=1&page[size]=30
   * @returns {string|*}
   */
  getPagination = () => {
    const { currentPage, itemsPerPage } = this.state
    return `?page[number]=${currentPage + 1}&page[size]=${itemsPerPage}`
  }

  /**
   * example : filter[instanceName]=centreon&filter[fingerprint]=abc
   * @returns {string|*}
   */
  getFilters = () => {
    const { filters } = this.state;

    return filters.reduce((acc, { columnName, value }) => {
      if (columnName == 'catalog_level' && value == '0') {
        acc.push(`filter[with_catalog]=false`)
      } else {
        acc.push(`filter[${columnName}]=${value}`)
      }
      return acc
    }, []).join('&')
  }

  getData = () => {
    const { token, displayPluginPacks } = this.props
    displayPluginPacks(this.getPagination(), this.getFilters(), token)
  }

  render = () => {
    const { columns, currentPage } = this.state
    const { items, itemsPerPage, totalPages, totalItems, ...props } = this.props

    const rows = items ?
      Object.keys(items).reduce((acc, key) => {
        acc[items[key].order] = items[key]
        return acc
      }, [])
      : []

    return (
      <PluginPacksList
        columns={columns}
        rows={rows}
        itemsPerPage={itemsPerPage}
        totalPages={totalPages}
        totalItems={totalItems}
        currentPage={currentPage}
        changeFilters={this.changeFilters}
        changeCurrentPage={this.changeCurrentPage}
        {...props}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {
    token: store.authReducer.token,
    items: store.pluginPacksReducer.items,
    totalPages: store.pluginPacksReducer.totalPages,
    totalItems: store.pluginPacksReducer.totalItems,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    cleanHeader: () => {
      return dispatch(cleanHeader())
    },
    replaceHeaderTitle: (title) => {
      return dispatch(replaceTitle(title))
    },
    displayPluginPacks: (pagination, filters, token) => {
      return dispatch(getPluginPacks(pagination, filters, token))
    },
  }
}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(PluginPacksListContainer))