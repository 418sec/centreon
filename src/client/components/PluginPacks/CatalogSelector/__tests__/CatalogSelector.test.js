/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import configureStore from 'redux-mock-store'

Enzyme.configure({ adapter: new Adapter() })

import { CatalogSelector } from 'Components/PluginPacks/CatalogSelector'

describe('CatalogSelector', () => {
  const mockStore = configureStore()
  const store = mockStore({})
  let component

  it('renders correctly', () => {
    component = shallow(
      <CatalogSelector
        store={store}
        level='2'
      />
    ).dive().dive().dive()
    expect(toJson(component)).toMatchSnapshot()
  })

  it('opens catalof form popin correctly', () => {
    let actions

    component = mount(
      <CatalogSelector
        store={store}
        level='2'
      />
    )
    component.find('div').simulate('click')
    actions = store.getActions()
    expect(actions[0].type).toBe('OPEN_CATALOG_FORM_POPIN')
  })
})