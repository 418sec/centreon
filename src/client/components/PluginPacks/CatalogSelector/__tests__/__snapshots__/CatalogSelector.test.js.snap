// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogSelector renders correctly 1`] = `
<WithStyles(Avatar)
  className="CatalogSelector-purpleAvatar-3"
  onClick={[Function]}
>
  2
</WithStyles(Avatar)>
`;
